src/script/lua_api/l_object.cpp

```diff
@@ -722,15 +722,15 @@ int ObjectRef::l_set_detach(lua_State *L)
 
 	int parent_id = 0;
 	std::string bone = "";
-	v3f position;
-	v3f rotation;
+	v3f position = v3f(0, 0, 0);
+	v3f rotation = v3f(0, 0, 0);
 	co->getAttachment(&parent_id, &bone, &position, &rotation);
 	ServerActiveObject *parent = NULL;
 	if (parent_id)
 		parent = env->getActiveObject(parent_id);
 
 	// Do it
-	co->setAttachment(0, "", v3f(0,0,0), v3f(0,0,0));
+	co->setAttachment(0, "", position, rotation);
```

Comment:
Can you be sure that position and rotation are still valid vectors after being passed by reference to `getAttachment`?